test/apps/iodemo/io_demo.cc

```diff
@@ -1309,12 +1309,8 @@ class DemoServer : public P2pDemoCommon {
         w->init(this, conn, msg->sn, msg->conn_id, iov,
                 &conn_stat.completions<IO_WRITE>());
         conn_stat.bytes<IO_WRITE>() += msg->data_size;
-        if (!ucx_am_is_rndv(data_desc)) {
-            conn->recv_am_data(NULL, 0, NULL, data_desc, w);
-        } else {
-            conn->recv_am_data((*iov)[0].buffer(), (*iov)[0].size(),
-                               (*iov)[0].memh(), data_desc, w);
-        }
+        conn->recv_am_data((*iov)[0].buffer(), (*iov)[0].size(),
```

Comment:
maybe use ucx_am_is_rndv() inside recv_am_data?

Reply:
it is done:
https://github.com/openucx/ucx/blob/7c336285d93316b6dcf373027564166eff2a5d58/test/apps/iodemo/ucx_wrapper.cc#L966